component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/record/AvroSchema.java

```diff
@@ -96,21 +105,63 @@ public Type getType() {
             if (entries != null) {
                 return entries;
             }
-            entries =
-                    getActualDelegate().getFields().stream().filter(it -> it.schema().getType() != NULL).map(field -> {
-                        final Type type = mapType(field.schema());
-                        final AvroSchema elementSchema = new AvroSchema(
-                                type == Type.ARRAY ? unwrapUnion(field.schema()).getElementType() : field.schema());
-                        // readProp(unwrapUnion(field.schema()), KeysForAvroProperty.LABEL) is not good location in my
-                        // view
-                        return new SchemaImpl.EntryImpl(field.name(), field.getProp(KeysForAvroProperty.LABEL), type,
-                                field.schema().getType() == UNION, field.defaultVal(), elementSchema, field.doc(),
-                                field.getProps());
-                    }).collect(toList());
+            entries = this
+                    .getNonNullFields() //
+                    .filter(f -> !AvroSchema.isMetadata(f)) // only data fields
+                    .map(this::fromAvro) //
+                    .collect(toList());
         }
         return entries;
     }
 
+    @Override
+    @JsonbTransient
+    public Stream<Entry> getAllEntries() {
+        if (this.metadataEntries == null) {
+            synchronized (this) {
+                if (entries != null) {
+                    this.metadataEntries = this
+                            .getNonNullFields() //
+                            .filter(AvroSchema::isMetadata) // only metadata fields
+                            .map(this::fromAvro) //
+                            .collect(Collectors.toList());
+                }
+            }
+        }
+
+        return Stream.concat(this.getEntries().stream(), this.metadataEntries.stream());
+    }
+
+    private Stream<Field> getNonNullFields() {
+        return getActualDelegate().getFields().stream().filter(it -> it.schema().getType() != NULL);
+    }
+
+    private static boolean isMetadata(final Field f) {
+        return f.aliases() != null && f.aliases().contains(AvroSchema.METADATA_ALIAS_NAME);
```

Comment:
I don't know the usage of `aliases`, what's the purpose?

Reply:
This is how we store the fact that a field is a metadata field in apache Avro record.